RTFD: View Layout and Restore feature

annapurna-gupta opened the pull request:
**Purpose of PR?**:

Fixes #2904

annapurna-gupta committed:
Save view settings on exit to view_settings.json (#2831)

Save view settings on application close  for top, side, linear, and table for flight tracks
Implemented restoring of views logic and added a test to cover the functionality.

annapurna-gupta committed:
Save view settings on exit to view_settings.json (#2831)

Save view settings on application close  for top, side, linear, and table for flight tracks
Implemented restoring of views logic and added a test to cover the functionality.

annapurna-gupta committed:
Add function to save view settings for an operation in database (#2861)

* Add function to save and restore view settings for an operation in database


---------

Co-authored-by: Matthias Riße <9308656+matrss@users.noreply.github.com>

annapurna-gupta committed:
Add function to save view settings for an operation in database (#2861)

* Add function to save and restore view settings for an operation in database


---------

Co-authored-by: Matthias Riße <9308656+matrss@users.noreply.github.com>

annapurna-gupta committed:
Feat: Implement view sharing in operations with Manage Views widget (#2893)

Adds functionality to share custom views between operation members. Includes a 'Manage Views' widget allowing users to browse and apply shared views, as well as options for renaming views before sharing.  Addresses the need for collaborative view configurations within an operation.

annapurna-gupta committed:
Feat: Implement view sharing in operations with Manage Views widget (#2893)

Adds functionality to share custom views between operation members. Includes a 'Manage Views' widget allowing users to browse and apply shared views, as well as options for renaming views before sharing.  Addresses the need for collaborative view configurations within an operation.

annapurna-gupta committed:
Add View Layout and Restoring documentation

annapurna-gupta committed:
Reorder SharedView __init__ params (#2905)

* Reorder SharedView __init__ params

* Enable CI for prepare** branches

ReimarBauer commented on docs/components.rst:
```diff
@@ -10,6 +10,7 @@ Components
    plugins
    mswms
    mscolab
+   View Layout and Restoring 
```
please keep all lower case characters for the file name

annapurna-gupta replied:
ok

ReimarBauer commented on docs/View Layout and Restoring .rst:
```diff
@@ -0,0 +1,45 @@
+
+Tips
+----
+- Always save your flighttrack or operation before closing the application to ensure views are persisted.
```
operation does not have a special save, each change creates a new revision.  There is no user involved for saving.

annapurna-gupta replied:
understood

ReimarBauer reviewed (changes requested):
see comments

annapurna-gupta committed:
apply suggested change

annapurna-gupta committed:
Error Fix

ReimarBauer commented:
We need to rebase the branch with develop

annapurna-gupta committed:
Merge branch 'prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta' into View-Layout-and-Restoring-documentation

ReimarBauer commented on docs/View Layout and Restoring .rst:
```diff
@@ -0,0 +1,44 @@
+   - Each flighttrack maintains its own view configuration.
+
+2. **Restoring Operation Views**
+   - When activating anoperation, the last opened views will be restored automatically.
```
an operation

annapurna-gupta replied:
okk

annapurna-gupta committed:
Fix operation view wording

annapurna-gupta commented:
@ReimarBauer **prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta** does not have few old commit.

ReimarBauer commented:
> @ReimarBauer **prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta** does not have few old commit.

on my rebase there was no merge conflict. I do a comparison and add them again.

annapurna-gupta commented:
> > @ReimarBauer **prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta** does not have few old commit.
> 
> on my rebase there was no merge conflict. I do a comparison and add them again.

okk sure!!

ReimarBauer commented:
> > > @ReimarBauer **prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta** does not have few old commit.
> > 
> > 
> > on my rebase there was no merge conflict. I do a comparison and add them again.
> 
> okk sure!!

looking with git log --all. 

we can run git cherry-pick `<hash>`

Do you have identified already some missing?

annapurna-gupta commented:
> > > > @ReimarBauer **prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta** does not have few old commit.
> > > 
> > > 
> > > on my rebase there was no merge conflict. I do a comparison and add them again.
> > 
> > 
> > okk sure!!
> 
> looking with git log --all.
> 
> we can run git cherry-pick `<hash>`
> 
> Do you have identified already some missing?

<img width="1350" height="421" alt="image" src="https://github.com/user-attachments/assets/e4c6ed15-56fd-4eba-acad-8e41e4d9f83d" />
 
look at the top 3 commits, they are not from this pr

ReimarBauer commented:
> look at the top 3 commits, they are not from this pr


Have you done yourself a merge or a rebase?

We don't see in this PR any other change, so I assume the changes are aready included.

I compared the commit messages and looked up the git log. They are in, with the right date compared to dervelop.

* Save view settings on exit to view_settings.json (#2831)
```
commit dd6a1a6c8a810c8a54f5b08fdd4bb694c8339e9f
Author: Annapurna Gupta <ganu48600@gmail.com>
Date:   Tue Aug 5 22:11:46 2025 +0530

    Save view settings on exit to view_settings.json (#2831)

    Save view settings on application close  for top, side, linear, and table for flight tracks
    Implemented restoring of views logic and added a test to cover the functionality.
```

* Add function to save view settings for an operation in database (#2861)

```
commit e9145f9b841382ecf2b266fe0723831907dbee3a
Author: Annapurna Gupta <ganu48600@gmail.com>
Date:   Tue Sep 16 21:26:31 2025 +0530

    Add function to save view settings for an operation in database (#2861)

    * Add function to save and restore view settings for an operation in database


    ---------

    Co-authored-by: Matthias Riße <9308656+matrss@users.noreply.github.com>

```

* Feat: Implement view sharing in operations with Manage Views widget (#2893)

```
commit 5cff1a9ce19036519f129c77b198dae56dd00abe
Author: Annapurna Gupta <ganu48600@gmail.com>
Date:   Mon Sep 29 16:03:24 2025 +0530

    Feat: Implement view sharing in operations with Manage Views widget (#2893)

    Adds functionality to share custom views between operation members. Includes a 'Manage Views' widget allowing users to browse and apply shared views, as well as options for renaming views before sharing.  Addresses the need for collaborative view configurations within an operation.
```

When something would be missing you would see this also in "Files Changed" But there only the two changed. 
I assume this comes from a merge instead a rebase.

annapurna-gupta commented:
> > look at the top 3 commits, they are not from this pr
> 
> Have you done yourself a merge or a rebase?
> 
> We don't see in this PR any other change, so I assume the changes are aready included.
> 
> I compared the commit messages and looked up the git log. They are in, with the right date compared to dervelop.
> 
> * Save view settings on exit to view_settings.json ([Save view settings on exit to view_settings.json #2831](https://github.com/Open-MSS/MSS/pull/2831))
> 
> ```
> commit dd6a1a6c8a810c8a54f5b08fdd4bb694c8339e9f
> Author: Annapurna Gupta <ganu48600@gmail.com>
> Date:   Tue Aug 5 22:11:46 2025 +0530
> 
>     Save view settings on exit to view_settings.json (#2831)
> 
>     Save view settings on application close  for top, side, linear, and table for flight tracks
>     Implemented restoring of views logic and added a test to cover the functionality.
> ```
> 
> * Add function to save view settings for an operation in database ([feat: Operation View Persistence #2861](https://github.com/Open-MSS/MSS/pull/2861))
> 
> ```
> commit e9145f9b841382ecf2b266fe0723831907dbee3a
> Author: Annapurna Gupta <ganu48600@gmail.com>
> Date:   Tue Sep 16 21:26:31 2025 +0530
> 
>     Add function to save view settings for an operation in database (#2861)
> 
>     * Add function to save and restore view settings for an operation in database
> 
> 
>     ---------
> 
>     Co-authored-by: Matthias Riße <9308656+matrss@users.noreply.github.com>
> ```
> 
> * Feat: Implement view sharing in operations with Manage Views widget ([Feat: Share View in an Operation #2893](https://github.com/Open-MSS/MSS/pull/2893))
> 
> ```
> commit 5cff1a9ce19036519f129c77b198dae56dd00abe
> Author: Annapurna Gupta <ganu48600@gmail.com>
> Date:   Mon Sep 29 16:03:24 2025 +0530
> 
>     Feat: Implement view sharing in operations with Manage Views widget (#2893)
> 
>     Adds functionality to share custom views between operation members. Includes a 'Manage Views' widget allowing users to browse and apply shared views, as well as options for renaming views before sharing.  Addresses the need for collaborative view configurations within an operation.
> ```
> 
> When something would be missing you would see this also in "Files Changed" But there only the two changed. I assume this comes from a merge instead a rebase.

I rebased the branch into develop.

ReimarBauer commented:
> > > look at the top 3 commits, they are not from this pr
> > 
> > 
> > Have you done yourself a merge or a rebase?
> > We don't see in this PR any other change, so I assume the changes are aready included.
> > I compared the commit messages and looked up the git log. They are in, with the right date compared to dervelop.
> > 
> > * Save view settings on exit to view_settings.json ([Save view settings on exit to view_settings.json #2831](https://github.com/Open-MSS/MSS/pull/2831))
> > 
> > ```
> > commit dd6a1a6c8a810c8a54f5b08fdd4bb694c8339e9f
> > Author: Annapurna Gupta <ganu48600@gmail.com>
> > Date:   Tue Aug 5 22:11:46 2025 +0530
> > 
> >     Save view settings on exit to view_settings.json (#2831)
> > 
> >     Save view settings on application close  for top, side, linear, and table for flight tracks
> >     Implemented restoring of views logic and added a test to cover the functionality.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * Add function to save view settings for an operation in database ([feat: Operation View Persistence #2861](https://github.com/Open-MSS/MSS/pull/2861))
> > 
> > ```
> > commit e9145f9b841382ecf2b266fe0723831907dbee3a
> > Author: Annapurna Gupta <ganu48600@gmail.com>
> > Date:   Tue Sep 16 21:26:31 2025 +0530
> > 
> >     Add function to save view settings for an operation in database (#2861)
> > 
> >     * Add function to save and restore view settings for an operation in database
> > 
> > 
> >     ---------
> > 
> >     Co-authored-by: Matthias Riße <9308656+matrss@users.noreply.github.com>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * Feat: Implement view sharing in operations with Manage Views widget ([Feat: Share View in an Operation #2893](https://github.com/Open-MSS/MSS/pull/2893))
> > 
> > ```
> > commit 5cff1a9ce19036519f129c77b198dae56dd00abe
> > Author: Annapurna Gupta <ganu48600@gmail.com>
> > Date:   Mon Sep 29 16:03:24 2025 +0530
> > 
> >     Feat: Implement view sharing in operations with Manage Views widget (#2893)
> > 
> >     Adds functionality to share custom views between operation members. Includes a 'Manage Views' widget allowing users to browse and apply shared views, as well as options for renaming views before sharing.  Addresses the need for collaborative view configurations within an operation.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > When something would be missing you would see this also in "Files Changed" But there only the two changed. I assume this comes from a merge instead a rebase.
> 
> I rebased the branch onto develop.

The commit messages in question have been added on Commits on Oct 2, 2025 

Your https://github.com/annapurna-gupta/MSS/tree/prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta
is shown to me as 

```
This branch is [24 commits ahead of](https://github.com/annapurna-gupta/MSS/compare/Open-MSS%3AMSS%3Aprepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta...prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta) 

and [29 commits behind](https://github.com/annapurna-gupta/MSS/compare/prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta...Open-MSS%3AMSS%3Aprepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta) Open-MSS/MSS:prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta.
```
and in yours you also find the changes from Commits on Oct 2, 2025

<img width="1006" height="383" alt="Bildschirmfoto 2026-01-02 um 15 47 51" src="https://github.com/user-attachments/assets/e48e1cfa-5c41-41a0-a0d6-eae3ea425031" />


Please sync it too.

annapurna-gupta committed:
Merge branch 'Open-MSS:develop' into View-Layout-and-Restoring-documentation

annapurna-gupta commented:
> > > > look at the top 3 commits, they are not from this pr
> > > 
> > > 
> > > Have you done yourself a merge or a rebase?
> > > We don't see in this PR any other change, so I assume the changes are aready included.
> > > I compared the commit messages and looked up the git log. They are in, with the right date compared to dervelop.
> > > 
> > > * Save view settings on exit to view_settings.json ([Save view settings on exit to view_settings.json #2831](https://github.com/Open-MSS/MSS/pull/2831))
> > > 
> > > ```
> > > commit dd6a1a6c8a810c8a54f5b08fdd4bb694c8339e9f
> > > Author: Annapurna Gupta <ganu48600@gmail.com>
> > > Date:   Tue Aug 5 22:11:46 2025 +0530
> > > 
> > >     Save view settings on exit to view_settings.json (#2831)
> > > 
> > >     Save view settings on application close  for top, side, linear, and table for flight tracks
> > >     Implemented restoring of views logic and added a test to cover the functionality.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > * Add function to save view settings for an operation in database ([feat: Operation View Persistence #2861](https://github.com/Open-MSS/MSS/pull/2861))
> > > 
> > > ```
> > > commit e9145f9b841382ecf2b266fe0723831907dbee3a
> > > Author: Annapurna Gupta <ganu48600@gmail.com>
> > > Date:   Tue Sep 16 21:26:31 2025 +0530
> > > 
> > >     Add function to save view settings for an operation in database (#2861)
> > > 
> > >     * Add function to save and restore view settings for an operation in database
> > > 
> > > 
> > >     ---------
> > > 
> > >     Co-authored-by: Matthias Riße <9308656+matrss@users.noreply.github.com>
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > * Feat: Implement view sharing in operations with Manage Views widget ([Feat: Share View in an Operation #2893](https://github.com/Open-MSS/MSS/pull/2893))
> > > 
> > > ```
> > > commit 5cff1a9ce19036519f129c77b198dae56dd00abe
> > > Author: Annapurna Gupta <ganu48600@gmail.com>
> > > Date:   Mon Sep 29 16:03:24 2025 +0530
> > > 
> > >     Feat: Implement view sharing in operations with Manage Views widget (#2893)
> > > 
> > >     Adds functionality to share custom views between operation members. Includes a 'Manage Views' widget allowing users to browse and apply shared views, as well as options for renaming views before sharing.  Addresses the need for collaborative view configurations within an operation.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > When something would be missing you would see this also in "Files Changed" But there only the two changed. I assume this comes from a merge instead a rebase.
> > 
> > 
> > I rebased the branch onto develop.
> 
> The commit messages in question have been added on Commits on Oct 2, 2025
> 
> Your https://github.com/annapurna-gupta/MSS/tree/prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta is shown to me as
> 
> ```
> This branch is [24 commits ahead of](https://github.com/annapurna-gupta/MSS/compare/Open-MSS%3AMSS%3Aprepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta...prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta) 
> 
> and [29 commits behind](https://github.com/annapurna-gupta/MSS/compare/prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta...Open-MSS%3AMSS%3Aprepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta) Open-MSS/MSS:prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta.
> ```
> 
> and in yours you also find the changes from Commits on Oct 2, 2025
> 
> <img alt="Bildschirmfoto 2026-01-02 um 15 47 51" width="1006" height="383" src="https://private-user-images.githubusercontent.com/278728/531494395-e48e1cfa-5c41-41a0-a0d6-eae3ea425031.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc0NjIyNzksIm5iZiI6MTc2NzQ2MTk3OSwicGF0aCI6Ii8yNzg3MjgvNTMxNDk0Mzk1LWU0OGUxY2ZhLTVjNDEtNDFhMC1hMGQ2LWVhZTNlYTQyNTAzMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEwM1QxNzM5MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMjE0YjU0MWZkNGVhN2I1OGU4NThjNzYwYzFlZDNmZjE3M2JjOWIxNWJlN2FhNDI1YzM3ZDc4N2FmYzQ0ZTcyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RJ0kq_w3JwwOsTXLEs-4DbRoETl5o0EEfUwOHVftNJY">
> Please sync it too.

i think i made a mistake, will fix it.

annapurna-gupta committed:
Merge branch 'Open-MSS:develop' into View-Layout-and-Restoring-documentation

annapurna-gupta commented:
@ReimarBauer I’ve updated my branch, but GitHub still shows some older commits in the comparison.
Could you please update **Open-MSS/MSS:prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupt**a as well.

ReimarBauer commented:
The PR has now unrelated changes to the issue. It should have only changes to the documentation.

ReimarBauer commented:
> @ReimarBauer I’ve updated my branch, but GitHub still shows some older commits in the comparison. Could you please update **Open-MSS/MSS:prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupt**a as well.

I rebased it on develop

ReimarBauer commented:
@annapurna-gupta maybe do the following. cherry pick the doc changes from this branch, rename the branch then
get your `prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta`  in sync with the one from `OpenMSS/MSS`. Then make an new branch for these changes name it as the old one based on your `prepare_merge_to_develop_from...`. Then add the changes and send it per fastforward to github.

annapurna-gupta commented:
> @annapurna-gupta maybe do the following. cherry pick the doc changes from this branch, rename the branch then get your `prepare_merge_to_develop_from_GSOC2025-AnnapurnaGupta` in sync with the one from `OpenMSS/MSS`. Then make an new branch for these changes name it as the old one based on your `prepare_merge_to_develop_from...`. Then add the changes and send it per fastforward to github.

Done, thanks
